src/libhoney.js

```diff
@@ -118,10 +118,17 @@ export default class Libhoney extends EventEmitter {
   }
 
   _responseCallback(responses) {
-    let queue = this._responseQueue;
-    if (queue.length < this._options.maxResponseQueueSize) {
-      this._responseQueue = this._responseQueue.concat(responses);
+    const [queue, limit] = [
+      this._responseQueue,
+      this._options.maxResponseQueueSize
+    ];
+
+    if (queue.length < limit) {
```

Comment:
I think this should do it, I do think this is easier to read. I noticed util-looking functions at the bottom of the doc, so I stuck the new function there, but lemme know if there's a place that makes more sense.

Reply:
I was suggesting that _all_ of the logic dealing with how to build the value of `this._responsesQueue` from the 3 parameters would go to `concatWithMaxLimit`, so this function could look like:

```
  _responseCallback(responses) {
    this._responseQueue = concatWithMaxLimit(this._responseQueue,
                                             responses,
                                             this.options.maxResponseQueueSize);
    this.emit("response", this._responseQueue);
  }
```

`concatWithMaxLimit` gains a very easy early-out at the top to handle the remaining conditional here:

```
  if (arr1.length >= limit) {
    return arr1;
  }
```